### ISSUE TITLE
[DOC] Inaccurate example(s) for register snapshots

### ISSUE THREAD

**Issue by JeffH-AWS** [opensearch-project/documentation-website#3028]
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [ ] Add new documentation
- [X] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
Seems to affect all versions (only bothering to look at 1.3 and newer):
e.g.:
- https://opensearch.org/docs/1.3/api-reference/snapshots/create-repository/
- https://opensearch.org/docs/2.5/api-reference/snapshots/create-repository/

I don't think we're defining the `base_path` parameter correctly, or if we are then we aren't explaining it completely.

In the description of `base_path` the example (my/snapshot/directory) uses a relative path:

```
The path within the bucket where you want to store snapshots (e.g. my/snapshot/directory). Optional. If not specified, snapshots are stored in the bucket root.
```

Same with the example:

```
PUT _snapshot/my-opensearch-repo
{
  "type": "s3",
  "settings": {
    "bucket": "my-open-search-bucket",
    "base_path": "my/snapshot/directory"
  }
}
```

The documentation needs to explain how the path defined here translates to the mounted volume being used. Is it relative to the WKDIR defined in the Dockerfile? But that would only apply to Docker images. Does it use an environment variable like OPENSEARCH_HOME?

I think it probably *must* be an absolute path, but since this needs corrected anyway let's find out what happens when you specify a relative path and include that on the page as well.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

I don't have any additional info - if help is needed we'll have to rope in one of the devs.

**Comment by JeffH-AWS** [comment]
Removed "1.3" from the title because I realized that it's the same across all the maintained versions.

**Comment by JeffH-AWS** [comment]
After a little further thought I think what might need to happen is that S3 and fs repositories should probably be explained separately unless the differences and distinctions are few enough to just call out differences as needed.

But we should have a clear understanding of two behaviors:

1. When you're using an S3 bucket as a snapshot repository, how does the path specified translate to `s3://<web-bucket>/`? In the example used would it look like `s3://my-open-search-bucket/my/snapshot/directory/`? I'm curious about how it concatenates the strings to form the URI.
2. When you're using a mounted fs volume for the repository, how is the path resolved? For example, if I mount a volume to `/mnt/snapshots` and then want to register a snapshot that would be stored in a directory called `some_directory` then what would the command look like?

I also noticed that we don't define `location` very well for the fs repo:
```
The shared file system for snapshots. Required.
```

That doesn't describe what the parameter does at all. Can we do a pass on required fields and make sure they're all defined clearly?

**Comment by ariamarble** [comment]
It looks like this is much more comprehensively documented at https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore/#register-repository. Thinking of the best way to integrate these together.

**Comment by ariamarble** [comment]
I can confirm that the steps and config parameters are correct as described in the walkthrough above. `base_path` only exists for the s3 config so the type of path is irrelevant. I also confirmed that when setting up an s3 repository that the `s3://` prefix is not required and I was able to successfully take snapshots to both s3 and a local path. What I'm going to do however since the overlap between the user guide and API reference is confusing, I'm going to restructure the two pages.
